Nonblocking Collectives (Draft) (#456)
openshmem-org/specification: content/shmem_barrier_all_nb.tex

\apisummary{
    Registers the arrival of a \ac{PE} at a barrier and returns immediately. It completes when all \acp{PE}
    arrive at the barrier and all local updates and remote memory updates on the default context are completed. 
    }

\begin{apidefinition}

\begin{Csynopsis}

> void @\FuncDecl{shmem\_barrier\_all\_nb}@(void);

Author: missing request object and tag

\apisummary{
    Registers the arrival of a \ac{PE} at a barrier and returns immediately. It completes when all \acp{PE}
    arrive at the barrier and all local updates and remote memory updates on the default context are completed. 
    }

\begin{apidefinition}

\begin{Csynopsis}
void @\FuncDecl{shmem\_barrier\_all\_nb}@(shmem_req_h *request);
\end{Csynopsis}

\begin{apiarguments}

    \apiargument{OUT}{request}{An opaque request handle identifying the
    collective operation.}

\end{apiarguments}

\apidescription{   
    Similar to the \FUNC{shmem\_barrier\_all} routine, the nonblocking \FUNC{shmem\_barrier\_all\_nb}
    is a mechanism for synchronizing all \acp{PE} in the world team at
    once. This routine completes when all \acp{PE} have called
    \FUNC{shmem\_barrier\_all\_nb}. 

>     A call to the nonblocking barrier routine posts the operation and returns
>     immediately without necessarily completing the operation. Upon successful posting of the operation, 

Reviewer: ```suggestion
    A call to the nonblocking barrier routine initiates the operation and returns
    immediately without necessarily completing the operation. On success, 
```
